pytensor/tensor/signal/conv.py

```diff
@@ -0,0 +1,170 @@
+from typing import TYPE_CHECKING, Literal, cast
+
+from numpy import convolve as numpy_convolve
+
+from pytensor.graph import Apply, Op
+from pytensor.scalar.basic import upcast
+from pytensor.tensor.basic import as_tensor_variable, join, zeros
+from pytensor.tensor.blockwise import Blockwise
+from pytensor.tensor.math import maximum, minimum
+from pytensor.tensor.type import vector
+from pytensor.tensor.variable import TensorVariable
+
+
+if TYPE_CHECKING:
+    from pytensor.tensor import TensorLike
+
+
+class Conv1d(Op):
+    __props__ = ("mode",)
+    gufunc_signature = "(n),(k)->(o)"
+
+    def __init__(self, mode: Literal["full", "valid"] = "full"):
+        if mode not in ("full", "valid"):
```

Comment:
I chose to exclude `same` from our representation because It doesn't seem very meaningful for rewrites. It's also odd if we have the numpy/scipy behavior variants. And the grad was a bit of a mess.

I considered only using one of full/valid but only full made some graphs rather wasteful (full conv followed by slicing away what we don't care about), whereas only `valid` ended up requiring padding/full all the time for default / gradient which was enough to convince me of the usefulness of having a direct full. Also most

Also most rewrites I can think of are more applicable to full than valid.

Reply:
This is all fair, but why handle the same implementation in the function, instead of in the op? It seems like it forces us to look around needlessly when we're looking for where the "same" logic is implemented. Could be done in the `make_node`, for example. Or just allow it in the forward, and rewrite to the padded representation if gradients are required.

Reply:
Why would it? Same is a concept the Op doesn't have to know about. Also we may implement a second numpy-like same, and the Op shouldn't have to know anything about it.

Reply:
Magic in make_node is fine for stuff like I had a float32 Op but now I want a float64. As a numpy-scipy helper bridge I find separate functions more clear. Otherwise we're mixing the implementation of a core Op with it's user facing applications